src/zenml/integrations/modal/orchestrators/modal_orchestrator.py

```diff
@@ -0,0 +1,206 @@
+
+    This orchestrator runs complete pipelines using Modal sandboxes
+    for maximum flexibility and efficiency with persistent app architecture.
+    """
+
+    @property
+    def config(self) -> "ModalOrchestratorConfig":
+        """Returns the Modal orchestrator config.
+
+        Returns:
+            The Modal orchestrator config.
+        """
+        return cast("ModalOrchestratorConfig", self._config)
+
+    @property
+    def settings_class(self) -> Optional[Type["BaseSettings"]]:
+        """Settings class for the Modal orchestrator.
+
+        Returns:
+            The settings class.
+        """
+        from zenml.integrations.modal.flavors.modal_orchestrator_flavor import (
+            ModalOrchestratorSettings,
+        )
+
+        return ModalOrchestratorSettings
+
+    def _setup_modal_client(self) -> None:
+        """Setup Modal client with authentication."""
+        setup_modal_client(
+            token_id=self.config.token_id,
+            token_secret=self.config.token_secret,
+            workspace=self.config.workspace,
+            environment=self.config.modal_environment,
+        )
+
+    @property
+    def validator(self) -> Optional[StackValidator]:
+        """Ensures there is a container registry and artifact store in the stack.
+
+        Returns:
+            A `StackValidator` instance.
+        """
+        validator: StackValidator = create_modal_stack_validator()
+        return validator
+
+    def get_orchestrator_run_id(self) -> str:
+        """Returns the active orchestrator run id.
+
+        Raises:
+            RuntimeError: If the environment variable specifying the run id
+                is not set.
+
+        Returns:
+            The orchestrator run id.
+        """
+        try:
+            return os.environ[ENV_ZENML_MODAL_ORCHESTRATOR_RUN_ID]
+        except KeyError:
+            raise RuntimeError(
+                "Unable to read run id from environment variable "
+                f"{ENV_ZENML_MODAL_ORCHESTRATOR_RUN_ID}."
+            )
+
+    def prepare_or_run_pipeline(
+        self,
+        deployment: "PipelineDeploymentResponse",
+        stack: "Stack",
+        environment: Dict[str, str],
+        placeholder_run: Optional["PipelineRunResponse"] = None,
+    ) -> Optional[Iterator[Dict[str, MetadataType]]]:
+        """Runs the complete pipeline using Modal sandboxes.
+
+        Args:
+            deployment: The pipeline deployment to prepare or run.
+            stack: The stack the pipeline will run on.
+            environment: Environment variables to set in the orchestration
+                environment.
+            placeholder_run: An optional placeholder run for the deployment.
+
+        Raises:
+            Exception: If pipeline execution fails.
+
+        Returns:
+            None if the pipeline is executed synchronously, otherwise an iterator of metadata dictionaries.
+        """
+        if deployment.schedule:
```

Comment:
We fail earlier when trying to run with a schedule, so no need to have this warning